patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -4961,15 +4961,19 @@
  			if (!(nPC.active & (nPC.shimmerTransparency == 0f || nPC.CanApplyHunterPotionEffects())))
  				continue;
  
-@@ -34825,6 +_,8 @@
+@@ -34825,8 +_,12 @@
  			if (nPC.type >= 87 && nPC.type <= 92)
  				value = new Microsoft.Xna.Framework.Rectangle((int)((double)nPC.position.X + (double)nPC.width * 0.5 - 32.0), (int)((double)nPC.position.Y + (double)nPC.height * 0.5 - 32.0), 64, 64);
  
 +			NPCLoader.ModifyHoverBoundingBox(nPC, ref value);
 +
  			bool flag = mouseRectangle.Intersects(value);
  			bool flag2 = flag || (SmartInteractShowingGenuine && SmartInteractNPC == i);
++			NPCLoader.OnHoverBoundingBox(nPC, value, flag, flag2);
```

Comment:
I'd rather this hook only be called `if (flag2)`, you can still pass `flag` as `mouseOver`

Reply:
wdym '`if (flag2)`'? inside the if statement below or its own if stamaent